doc/type_requirements/File.qbk

```diff
@@ -0,0 +1,183 @@
+        lowest byte offset for which a read will always return a
+        [link nudb.ref.nudb__error short_read] error. Behavior is undefined
+        if `f` does not refer to an open file.
+    ]
+]
+[
+    [`f.read(o,a,n,ec)`]
+    [`void`]
+    [
+        Attempts to read `n` bytes starting at file offset `o`
+        from the open file referred to by `f`. Bytes read are stored
+        in the memory buffer at address 'a'. The memory buffer at 'a'
+        must be at least n bytes long. If an error occurs, `ec` is set
+        to the system specific error code. Behavior is undefined if `f`
+        does not refer file opened with [link nudb.ref.nudb__file_mode file_mode]
+        equal to scan or read.
+    ]
+]
+[
+    [`f.sync(ec)`]
+    [`void`]
+    [
+        Attempts to synchronize the file on disk. This instructs the
+        operating system to ensure that any data which resides in caches
+        or buffers is fully written to the underlying storage device
+        before this call returns. If an error occurs, `ec` is set to the
+        system specific error code. Behavior is undefined if
+        `f` does not refer to an open file.
+
+        NuDB's database integrity guarantees are only valid if the
+        implementation of `sync` assures that all data is fully written
+        to the underlying file before the call returns.
+    ]
+]
+[
+    [`f.write(o,a,n,ec)`]
+    [`void`]
+    [
+        Attempts to write `n` bytes starting at file offset `o`
+        to the open file referred to by `f`.  The memory buffer at
+        `a` should contain the `n` bytes meant to be copied to the file.
+        If an error occurs, `ec` is set to the system specific error code.
+        Behavior is undefined if `f` does not refer a file opened with
+        [link nudb.ref.nudb__file_mode file_mode] equal to write or append.
+    ]
+]
+[
+    [`f.trunc(o,ec)`]
+    [`void`]
+    [
+        Attempts to change the size of the open file referred to by `f`
+        to the size in bytes specified by `o`. If an error occurs, `ec`
+        is set to the system specific error code. Behavior is undefined if
+        `f` does not refer to an open file. After a successful call,
+        `a.size(ec)` will return `o`.
+
+        NuDB's database integrity guarantees are only valid if the
+        implementation of `trunc` assures that subsequent calls to
+        `size` will return `o`, even if the program is terminated or the
+        device is taken offline before calling `size`.
+    ]
+]
+[
+    [`is_File<F>`]
+    [`std::true_type`]
+    [
+        returns `std::true_type` for `F`. Otherwise returns `std::false_type`
+    ]
+]
+]
+
+[heading Associated Functions]
+* [link nudb.ref.nudb__block_size block_size]
+* [link nudb.ref.nudb__erase_file erase_file]
+
+[heading Models]
+* [link nudb.ref.nudb__posix_file posix_file]
+* [link nudb.ref.nudb__win32_file win32_file]
+* [link nudb.ref.nudb__native_file native_file]
+[br]
```

Comment:
I just replaced it with a newline and it seemed to work out okay in the generated docs.